scripts/setup_osx.sh

```diff
@@ -1,38 +1,265 @@
+# THE DEFAULTS INITIALIZATION - OPTIONALS
+_arg_set_envvar="off"
 
-# Make sure that we're under the directory of the project
-cd "$(dirname "$0")/.."
+print_help ()
+{
+  printf '%s\n' "Installer of garage for OS X."
+  printf 'Usage: %s [--(no-)set-envvar] ' "$0"
+  printf '[-h|--help] <mjkey-path>\n' "$0"
+  printf '\t%s\n' "<mjkey-path>: Path of the MuJoCo key"
+  printf '\t%s' "--set-envvar,--no-set-envvar: Set environment variables "
+  printf '%s\n' "required by garage in .bash_profile (off by default)"
+  printf '\t%s\n' "-h,--help: Prints help"
+}
 
-echo "Setting up MuJoCo..."
-/bin/bash ./scripts/setup_mujoco.sh
-if [[ "${?}" -ne 0 ]]; then
-  echo -e "\e[0;31mError: MuJoCo couldn't be set up\e[0m"
-  exit 1
+parse_commandline ()
+{
+  while test $# -gt 0
+  do
+    _key="$1"
+    case "$_key" in
+      --no-set-envvar|--set-envvar)
+        _arg_set_envvar="on"
+        test "${1:0:5}" = "--no-" && _arg_set_envvar="off"
+        ;;
+      -h|--help)
+        print_help
+        exit 0
+        ;;
+      -h*)
+        print_help
+        exit 0
+        ;;
+      *)
+        _positionals+=("$1")
+        ;;
+    esac
+    shift
+  done
+}
+
+
+handle_passed_args_count ()
+{
+  _required_args_string="'mjkey-path'"
+  error_msg="$(echo "FATAL ERROR: Not enough positional arguments - we" \
+    "require exactly 1 (namely: $_required_args_string), but got only" \
+    "${#_positionals[@]}.")"
+  test ${#_positionals[@]} -ge 1 || _PRINT_HELP=yes die "${error_msg}" 1
+  error_msg="$(echo "FATAL ERROR: There were spurious positional arguments" \
+    "--- we expect exactly 1 (namely: $_required_args_string), but got" \
+    "${#_positionals[@]} (the last one was: '${_positionals[*]: -1}').")"
+  test ${#_positionals[@]} -le 1 || _PRINT_HELP=yes die "${error_msg}" 1
+}
+
+assign_positional_args ()
+{
+  _positional_names=('_arg_mjkey_path' )
+
+  for (( ii = 0; ii < ${#_positionals[@]}; ii++))
+  do
+    error_msg="$(echo "Error during argument parsing, possibly an Argbash" \
+      "bug.")"
+    eval "${_positional_names[ii]}=\${_positionals[ii]}" || \
+      die "${error_msg}" 1
+  done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+
+# Utility functions
+script_dir_path() {
+  SCRIPT_DIR="$(dirname ${0})"
+  [[ "${SCRIPT_DIR}" = /* ]] && echo "${SCRIPT_DIR}" || \
+    echo "${PWD}/${SCRIPT_DIR#./}"
+}
+
+# red text
+print_error() {
+  echo -e "\033[0;31m${@}\033[0m"
+}
+
+# yellow text
+print_warning() {
+  echo -e "\033[0;33m${@}\033[0m"
+}
+
+# List of verified versions of OS X where garage has been installed
+VALID_VER=()
+VER="$(sw_vers -productVersion)"
+
+if [[ ! " ${VALID_VER[@]} " =~ " ${VER} " ]]; then
+  print_warning "It has not been verified whether garage will install" \
+    "correctly under your current OS X version (${VER}). If the installation" \
+    "is successful, please update the list of valid versions in this script." \
```

Comment:
1. Provide a link to this file on Github
1. Rename `VALID_VER` to `VERIFIED_OSX_VERSIONS` to make it more clear to contributors what to edit
1. Put `VERIFIED_OSX_VERSIONS` at the top of the file, and leave a block comment explaining that this is the list you should edit if you were directed here by the installer
> You are attempting to install garage on a version of OS X which we have not verified is working.
>
> garage relies on community contributions to support OS X
>
> If this installation is successful, please add your OS X version to `VERIFIED_OSX_VERSIONS` to https://github.com/rlworkgroup/garage/blob/master/scripts/setup_osx.sh on GitHub and submit a pull request to rlworkgroup/garage to help out future users. If the installation is not initially successful, but you find changes which fix it, please help us out by submitting a PR with your updates to the setup script.